tests/hostpath_quota_test.go

```diff
@@ -0,0 +1,606 @@
+					BeNil(),
+					"while deleting pod {%s} in namespace {%s}",
+					podName,
+					namespaceObj.Name,
+				)
+
+				By("verifying pod count as 0")
+				podCount := ops.GetPodRunningCountEventually(namespaceObj.Name, label, 0)
+				Expect(podCount).To(Equal(0), "while verifying pod count")
+
+				By("deleting the PVC with StorageClass " + scName)
+				pvName := ops.GetPVNameFromPVCName(pvcName)
+				Expect(pvName).ToNot(
+					BeEmpty(),
+					"while getting Spec.VolumeName from "+
+						"PVC {%s} in namespace {%s}",
+					pvcName,
+					namespaceObj.Name,
+				)
+
+				By("deleting above PVC")
+				err = ops.PVCClient.Delete(context.TODO(), pvcName, &metav1.DeleteOptions{})
+				Expect(err).To(
+					BeNil(),
+					"while deleting pvc {%s} in namespace {%s}",
+					pvcName,
+					namespaceObj.Name,
+				)
+
+				By("verifying PVC is deleted")
+				status := ops.IsPVCDeletedEventually(pvcName, namespaceObj.Name)
+				Expect(status).To(
+					BeTrue(),
+					"when checking status of deleted PVC {%s}",
+					pvcName,
+				)
+
+				By("deleting the storageClass " + scName)
+				err = ops.SCClient.Delete(context.TODO(), scName, &metav1.DeleteOptions{})
+				Expect(err).To(
+					BeNil(),
+					"while deleting storageclass {%s}",
+					scName,
+				)
+			})
+		})
+	})
+
+	When("Destroying the disk", func() {
+		It("should dettach the loopback device and delete the backing image", func() {
+			DestroyDisk(physicalDisk)
+		})
+	})
+})
+
+// BuildPersistentVolumeClaim builds the PVC object
+func BuildPersistentVolumeClaim(pvcName, scName, capacity string, accessModes []corev1.PersistentVolumeAccessMode) (*corev1.PersistentVolumeClaim, error) {
+	return pvc.NewBuilder().
+		WithName(pvcName).
+		WithNamespace(namespaceObj.Name).
+		WithStorageClass(scName).
+		WithAccessModes(accessModes).
+		WithCapacity(capacity).Build()
+}
+
+// BuildPod builds the pod object
+func BuildPod(podName, pvcName string, labelselector map[string]string) (*corev1.Pod, error) {
+	return pod.NewBuilder().
+		WithName(podName).
+		WithNamespace(namespaceObj.Name).
+		WithLabels(labelselector).
+		WithContainerBuilder(
+			container.NewBuilder().
+				WithName("busybox").
+				WithImage("busybox").
+				WithCommandNew(
+					[]string{
+						"sleep",
+						"3600",
+					},
+				).
+				WithVolumeMountsNew(
+					[]corev1.VolumeMount{
+						{
+							Name:      "demo-vol1",
+							MountPath: "/mnt/store1",
+						},
+					},
+				),
+		).
+		WithVolumeBuilder(
+			volume.NewBuilder().
+				WithName("demo-vol1").
+				WithPVCSource(pvcName),
+		).
+		Build()
+}
+
+// PrepareDisk prepares the setup necessary for testing xfs hostpath quota
+func PrepareDisk(fsType, hostPath string) disk.Disk {
+	xfsHostpathDir := "/var/openebs/integration-test/xfs/"
+	physicalDisk := disk.NewDisk(DiskImageSize)
+
+	fmt.Println("Calling attach function")
+	err := physicalDisk.AttachDisk()
+	Expect(err).To(
+		BeNil(),
+		"while creating loopback device with disk {%+v}",
+		physicalDisk,
+	)
```

Comment:
As a best practice let's not use gomega assertions in functions... Let's return error if there is any error and if caller is for test then we can have assertions over there... 
```suggestion
	if err != nil {
	    return errors.Wrapf(err, "while creating loop back device with disk %+v", physicalDisk)
	}
```